src/generator/doc_repo.cr

```diff
@@ -136,11 +145,23 @@ module Generator
                      elsif identifier.starts_with?("set_") && identifier.size > 4
                        "##{identifier[4..]}="
                      else
-                       "##{identifier}"
+                       "#{split_reference.size == 2 ? "::" : "#"}#{identifier}"
                      end
 
         "`#{split_reference[0..-2].join("::")}#{identifier}`"
       end
+
+      crystallized_doc = crystallized_doc.gsub(/(G[dts]k|Pango|cairo_|graphene_|Adw|Hdy|GtkSource)(\w+\b)/) do |_, match|
+        part_1 = match[-2]?
+        part_2 = match[-1]?
+
+        unless part_1.nil? || part_2.nil?
+          "#{part_1}::#{part_2}"
+        else
+          match[0]
+        end
```

Comment:
This block is only called when the regex matches, so it's safe to just join the captures.
```suggestion
        match.captures.join("::")
```